library/ziti.c

```diff
@@ -1052,23 +1118,26 @@ static void session_cb(ziti_session *session, const ziti_error *err, void *ctx)
     int errCode = err ? err->err : ZITI_OK;
 
     if (session) {
-        ZTX_LOG(DEBUG, "%s successfully => api_session[%s]", ztx->session ? "refreshed" : "logged in", session->id);
+        ZTX_LOG(DEBUG, "%s successfully => api_session[%s]", ztx->api_session ? "refreshed" : "logged in", session->id);
 
-        set_session(ztx, session);
+        ziti_set_api_session(ztx, session);
 
         ziti_auth_query_init(ztx);
 
+        ztx->api_session_state = ZitiApiSessionStatePartiallyAuthenticated;
```

Comment:
if I read this correctly `ziti_set_api_session()` may set state to fully auth'ed if there are no auth_queries, 
but then this line will set it back to partial auth'ed

is this correct?

Reply:
looks like it is already done there

Reply:
This was changed. It is now a single call to `ziti_set_api_session` which in turn uses the state transition functions which provide logging. `ziti_set_api_session` detect partial/full auth based on outstanding auth queries.